### PR TITLE
Add host address binding option

### PR DESCRIPTION
**Pull request by jkcoxson**
This PR allows for the user to define the socket address that onetun should bind to. This allows for systems where the outgoing address on the other end is not modifiable (such is the case on the iOS and Android client).

### CODE REVIEW

File: src/config.rs
Comment (aramperes) [inline src/config.rs:85]:
```suggestion
                    .help("The address (IP + port) used to bind the local UDP socket for the WireGuard tunnel. Example: 1.2.3.4:30000. Defaults to 0.0.0.0:0 for IPv4 endpoints, or [::]:0 for IPv6 endpoints."),
```

File: src/config.rs
Comment (aramperes) [inline src/config.rs:249]: This should additionally check that the IP version of the host address matches the version of the endpoint address (IPv4/IPv6).

File: src/config.rs
Comment (aramperes) [inline src/config.rs:80]: I'm tempted to make the name more explicit, such as `endpoint-bind-addr`. Thoughts?

**Review by aramperes — CHANGES_REQUESTED** [review]
Thanks for the PR! I have some suggestions before merging this.

File: src/config.rs
Comment (jkcoxson) [reply src/config.rs:80]: It's more descriptive if you think about it, but can also be more confusing when we already have ``endpoint_addr``. I'll leave it up to you; if you think it's a better idea I'll change it.

File: src/wg.rs
Comment (aramperes) [inline src/wg.rs:44]: Would make more sense for this logic to be moved up to `config.rs`, and make `Config#host_addr` a `SocketAddr` instead of `Option`.

File: src/config.rs
Comment (aramperes) [reply src/config.rs:80]: Yeah I'd prefer `endpoint-bind-addr`. Thanks!

**Comment by jkcoxson** [comment]
I have adjusted my PR according to your changes, let me know if you have any more comments.

File: src/config.rs
Comment (aramperes) [inline src/config.rs:242]: The logic here is inverted. Simply,

```suggestion
            if addr.ip().is_ipv4() != endpoint_addr.ip().is_ipv4()
```

**Review by aramperes — CHANGES_REQUESTED** [review]

File: src/config.rs
Comment (aramperes) [inline src/config.rs:245]:
```suggestion
                    "Endpoint and bind addresses must be the same IP version"
```

File: src/config.rs
Comment (aramperes) [inline src/config.rs:239]:
```suggestion
            let addr = parse_addr(Some(addr)).with_context(|| "Invalid bind address")?;
```

**Review by aramperes — APPROVED** [review]
Thank you!

**Pull request merged**
